### ISSUE TITLE
Tied lm_head bypasses SpyreParallelLMHead, so large-vocab models blow the per-core span

### ISSUE THREAD

**Issue by tdoublep** [torch-spyre/spyre-inference#631]
Models whose `lm_head` is tied to the embedding table die at generate once the table is large enough:

```
Unsupported: Spyre backend does not support: buf1: per-core tensor span 296.750 MB
(shape=[151936, 2048], dtype=torch.float16, device_size=[2374, 2048, 64],
splits={d0: 2, d1: 4}) exceeds hardware limit of 256.00 MB
```

## Cause

`tie_word_embeddings` makes Qwen2/Qwen3 assign `lm_head = self.model.embed_tokens` (`qwen3.py:299`), so the head is a `VocabParallelEmbedding` and `SpyreParallelLMHead` is never constructed. Logits fall through to `UnquantizedEmbeddingMethod.apply`'s `F.linear(x, weight)`, which has to materialize the `[vocab, hidden]` table vocab-innermost — the opposite of the row-gathered layout `SpyreVocabParallelEmbedding` places it in — as `buf1`, on every logits call.

Work division splits a stick dim only by a divisor of its stick count, subject to `prod(splits) <= sencores` (`work_division.py:must_split_vars`). A 151936 vocab is 2374 sticks = 2 * 1187, so 2 is the only admissible split: one core holds 1187 sticks, and the span passes 256 MB once hidden reaches 2048.

| Model | tied | hidden | span | result |
|---|---|---|---|---|
| Qwen3-0.6B | yes | 1024 | 148 MB | works, relayout each step |
| Qwen2.5-1.5B-Instruct | yes | 1536 | 223 MB | works, relayout each step |
| Qwen2.5-3B-Instruct | yes | 2048 | 297 MB | fails |
| Qwen3-1.7B | yes | 2048 | 297 MB | fails |
| Qwen3-4B | yes | 2560 | 371 MB | fails |
| Qwen2.5-7B-Instruct | no | 3584 | n/a | works |

The untied path is unaffected: `SpyreUnquantizedLMHeadMethod` pads vocab to a multiple of `64 * 32` and stores `Wᵀ`, so its stick count always has an admissible split. A tied head never reaches it.

## Scope

Not Qwen-specific. At least 21 model files assign the embedding directly (`olmo3`, `opt`, `bloom`, `falcon`, `mamba`, `seed_oss`, ...); the 26 that tie a real `ParallelLMHead` through `tie_weights()` are fine. Models below the span limit still pay the relayout on every step.

CI stays green because the allowlisted tied model, Qwen2.5-0.5B-Instruct, is hidden 896.

Fixed by #633.
